### PR TITLE
Convert pivot facets to a component and add hierarchy expand/collapse

### PR DESCRIPTION
**Pull request by cbeer**
Screen Shot 2020-06-04 at 10 01 15

### CODE REVIEW

File: app/components/blacklight/facet_item_pivot_component.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    def has_items?
+      @facet_item.items.present?
+    end
+
+    def facet_toggle_button
+      content_tag 'button', class: 'btn facet-toggle-handle collapsed',
+                            data: { toggle: 'collapse', target: "##{@id}" },
+                            aria: { expanded: false, controls: @id, describedby: "#{@id}_label" } do
+        concat content_tag 'span', t('blacklight.search.facets.pivot.show_html'), class: 'show'
```
Comment (jcoyne) [inline app/components/blacklight/facet_item_pivot_component.rb:50]: I'm really not a big fan of putting html in i18n. Is there a reason we can't put just the text of the  .sr-only element in the i18n file?

File: app/components/blacklight/facet_item_pivot_component.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# frozen_string_literal: true
+
+module Blacklight
+  # Render facet items and any subtree
+  class FacetItemPivotComponent < ::ViewComponent::Base
+    def self.id_counter
```
Comment (jcoyne) [inline app/components/blacklight/facet_item_pivot_component.rb:6]: Can you explain this counter? Does this just go up without bounds? Why do we need it?

File: config/locales/blacklight.de.yml [hunk 3]
```diff
@@ -194,6 +194,9 @@ de:
         selected:
           remove: '[entfernen]'
         missing: [fehlt]
+        pivot:
+          show:
+          hide_html: ⊟ <span class="sr-only"></span>
```
Comment (jcoyne) [inline config/locales/blacklight.de.yml:199]: Need to fix this one?

File: app/components/blacklight/facet_item_pivot_component.rb (hunk 2 above)
Comment (cbeer) [reply app/components/blacklight/facet_item_pivot_component.rb:6]: ARIA requires a unique ID to match the control with the thing it controls. I'm open to better ideas (blacklight-hierarchy uses `SecureRandom.uuid`, which is slow and.. unnecessary). 

Even at 1,000,000,000,000 cycles, the base 36 string is just 7 characters long, but I think most production servers will recycle threads well before that.

File: config/locales/blacklight.de.yml (hunk 3 above)
Comment (cbeer) [inline config/locales/blacklight.de.yml:199]:
```suggestion
          hide:
```

File: app/components/blacklight/facet_item_pivot_component.rb (hunk 2 above)
Comment (jcoyne) [reply app/components/blacklight/facet_item_pivot_component.rb:6]: That's super helpful context. Can you please add this information to a comment in this class?

Comment (cbeer) [reply app/components/blacklight/facet_item_pivot_component.rb]: Sure; and I've gone ahead and capped the counter at ~1 million, and made sure we only add the id  to leaf nodes. If you have 1 million leaf nodes on  a  page, you might have a problem, but it probably isn't markup validity, accessibility or wonky javascript behavior.

**Pull request merged**